File: android/app/src/main/kotlin/com/automattic/wordpresstv/home/FeaturedPromoBanner.kt
package com.automattic.wordpresstv.home

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.automattic.wordpresstv.R
import com.automattic.wordpresstv.promo.CodeForThePeople
import com.automattic.wordpresstv.promo.FeaturedBadge
import androidx.tv.material3.Card
import androidx.tv.material3.CardDefaults
import androidx.tv.material3.Text

/** The dark, faintly-branded ground the still fades into on its left. */
private val PromoNavy = Color(0xFF0B1636)

/**
 * The Home hero: a full-width featured banner for the "Code for the People"
 * documentary, above the rails. A cinematic still bleeds full-height to the right
 * edge and dissolves into [PromoNavy] on the left, so our own title/tagline read
 * over the dark side. NOTE: the intended art is a title-free still — the film's
 * title lives in the copy here, not baked into the image. The whole card is one
 * focusable control that opens [com.automattic.wordpresstv.promo.PromoScreen].
 * Mirrors the Apple `FeaturedPromoBanner`.
 */
@Composable
fun FeaturedPromoBanner(
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Card(
        onClick = onClick,
        modifier = modifier.fillMaxWidth().height(190.dp),
        scale = CardDefaults.scale(focusedScale = 1.02f),
    ) {
        Box(Modifier.fillMaxSize().background(PromoNavy)) {
            Image(
                painter = painterResource(R.drawable.code_for_the_people),
                contentDescription = CodeForThePeople.TITLE,
                contentScale = ContentScale.Crop,
                modifier = Modifier.align(Alignment.CenterEnd).fillMaxHeight().fillMaxWidth(0.64f),
            )

            Box(
                Modifier.fillMaxSize().background(
                    Brush.horizontalGradient(
                        0.0f to PromoNavy,
                        0.36f to PromoNavy,
                        0.66f to Color.Transparent,
                    ),
                ),
            )

            Column(
                modifier = Modifier
                    .align(Alignment.CenterStart)
                    .fillMaxWidth(0.48f)
                    .padding(start = 36.dp, end = 8.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp),
            ) {
                // The film's title treatment lives in the art on the right, so the
                // copy here doesn't repeat it — a short synopsis and the Watch cue.
                FeaturedBadge()
                Text(
                    text = stringResource(R.string.promo_home_synopsis),
                    color = Color.White.copy(alpha = 0.82f),
                    fontSize = 16.sp,
                    lineHeight = 22.sp,
                    fontWeight = FontWeight.Medium,
                )
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(12.dp),
                ) {
                    PlayBadge()
                    Text(
                        text = stringResource(R.string.promo_home_cta),
                        color = Color.White,
                        fontSize = 16.sp,
                        fontWeight = FontWeight.SemiBold,
                    )
                }
            }
        }
    }
}

/** A small circular play affordance next to the call to action. */
@Composable
private fun PlayBadge() {
    Box(
        modifier = Modifier.size(30.dp).clip(CircleShape).background(Color.White.copy(alpha = 0.16f)),
        contentAlignment = Alignment.Center,
    ) {
        Canvas(Modifier.size(12.dp)) {
            val path = Path().apply {
                moveTo(size.width * 0.12f, 0f)
                lineTo(size.width * 0.12f, size.height)
                lineTo(size.width, size.height / 2f)
                close()
            }
            drawPath(path, Color.White)
        }
    }
}

File: apple/WordPressTV/Home/FeaturedPromoBanner.swift
import SwiftUI

/// The Home hero: a full-width featured banner for the "Code for the People"
/// documentary, above the rails. A cinematic still bleeds full-height to the right
/// edge and dissolves into `navy` on the left, so our own title/tagline read over
/// the dark side. NOTE: the intended art is a title-free still — the film's title
/// lives in the copy here, not baked into the image. The whole card is one
/// focusable control that opens `PromoView`. Mirrors the Android
/// `FeaturedPromoBanner`.
struct FeaturedPromoBanner: View {
    let onOpen: () -> Void

    @FocusState private var focused: Bool

    /// The dark, faintly-branded ground the still fades into on its left.
    private static let navy = Color(red: 0.043, green: 0.086, blue: 0.212) // #0B1636
    // Tall enough that the ~1.85:1 still isn't over-cropped vertically — keeps the
    // banner near the Android hero's ~4.5:1 proportions so the title art reads.
    private static let height: CGFloat = 390

    var body: some View {
        Button(action: onOpen) { banner }
            .buttonStyle(PromoBannerButtonStyle(isFocused: focused))
            .focused($focused)
    }

    private var banner: some View {
        GeometryReader { geo in
            ZStack(alignment: .leading) {
                Self.navy

                // The still bleeds full-height to the right edge...
                Image("CodeForThePeople")
                    .resizable()
                    .scaledToFill()
                    .frame(width: geo.size.width * 0.64, height: geo.size.height)
                    .clipped()
                    .frame(maxWidth: .infinity, alignment: .trailing)

                // ...and dissolves into the navy on the left so the copy reads over it.
                LinearGradient(
                    stops: [
                        .init(color: Self.navy, location: 0.0),
                        .init(color: Self.navy, location: 0.36),
                        .init(color: .clear, location: 0.66),
                    ],
                    startPoint: .leading,
                    endPoint: .trailing
                )

                // The film's title ("Code for the People") lives in the art on the
                // right, so the copy here leads with the tagline and the Watch cue.
                copy
                    .frame(width: geo.size.width * 0.5, alignment: .leading)
                    .padding(.leading, 48)
            }
            .frame(width: geo.size.width, height: geo.size.height)
        }
        .frame(height: Self.height)
        .clipShape(RoundedRectangle(cornerRadius: 20))
    }

    private var copy: some View {
        VStack(alignment: .leading, spacing: 16) {
            FeaturedBadge()

            Text(CodeForThePeople.tagline)
                .font(.title3.weight(.medium))
                .foregroundStyle(.white.opacity(0.82))
                .fixedSize(horizontal: false, vertical: true)

            HStack(spacing: 14) {
                PlayBadge()
                Text("Watch the documentary")
                    .font(.headline)
                    .foregroundStyle(.white)
            }
        }
    }
}

/// A small circular play affordance next to the call to action.
private struct PlayBadge: View {
    var body: some View {
        Image(systemName: "play.fill")
            .font(.system(size: 15, weight: .bold))
            .foregroundStyle(.white)
            .frame(width: 40, height: 40)
            .background(Circle().fill(.white.opacity(0.16)))
    }
}

/// The banner's focus look: a subtle lift, ring, and shadow when the tvOS focus
/// engine highlights it — matching the `.card` feel without the poster crop. Driven
/// by an explicit `isFocused` (from the view's `@FocusState`), the same pattern the
/// nav-bar tab styles use.
private struct PromoBannerButtonStyle: ButtonStyle {
    var isFocused: Bool

    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .overlay(
                RoundedRectangle(cornerRadius: 20)
                    .stroke(.white.opacity(isFocused ? 0.9 : 0), lineWidth: 4)
            )
            .scaleEffect(isFocused ? 1.02 : 1)
            .shadow(color: .black.opacity(isFocused ? 0.5 : 0), radius: 24, y: 12)
            .opacity(configuration.isPressed ? 0.85 : 1)
            .animation(.easeOut(duration: 0.18), value: isFocused)
    }
}
